Assets/Scripts/Interactables/Resource/Resource.cs

```diff
@@ -0,0 +1,113 @@
+using Examen.Inventory;
+using Examen.Items;
+using Examen.Poolsystem;
+using FishNet.Object;
+using MarkUlrich.Health;
+using UnityEngine;
+
+namespace Examen.Interactables.Resource
+{
+    [RequireComponent(typeof(HealthData))]
+    public class Resource : NetworkBehaviour, Interactable
+    {
+        public Item ResourceItem;
+
+        [SerializeField] protected int p_amountToGive = 1;
+        [SerializeField] protected int p_damageAmount = 1;
+        [SerializeField] protected int p_deathTime;
+
+        protected HealthData p_healthData;
+        protected PoolSystem p_poolSystem;
+        protected bool p_hasHealthData;
+
+        /// <summary>
+        /// Sets PoolSystem instance.
+        /// </summary>
+        public virtual void Start() => p_poolSystem = PoolSystem.Instance;
+
+        /// <summary>
+        /// If object isServer, it resurrects this gameobject and calls for the clients to mimic its position and active state.
+        /// </summary>
+        public virtual void OnEnable()
+        {
+            if (!IsServer)
+                return;
+
+            if (!p_hasHealthData)
+            {
+                p_healthData = GetComponent<HealthData>();
+                p_hasHealthData = true;
+
+                p_healthData.onDie.AddListener(StartRespawnTimer);
+                p_healthData.onDie.AddListener(ToggleGameobject);
+
+                p_healthData.onResurrected.AddListener(ToggleGameobject);
+            }
+
+            SetNewPostion(transform.position);
+
+            p_healthData.Resurrect(p_healthData.MaxHealth);
+        }
+
+        /// <summary>
+        /// Starts the deathTimer and despawns this object.
+        /// </summary>
+        public virtual void StartRespawnTimer()
+        {
+            p_poolSystem.StartRespawnTimer(p_deathTime, ResourceItem.Name, transform.parent);
+            p_poolSystem.DespawnObject(ResourceItem.Name, gameObject);
+        }
+
+        /// <summary>
+        /// Set the active to the opposite of its current active state.
+        /// </summary>
+        [ObserversRpc]
+        public void ToggleGameobject() => gameObject.SetActive(!gameObject.activeSelf);
+
+        /// <summary>
+        /// Sets client resource position to server resource postion.
+        /// </summary>
+        /// <param name="serverPosition"> the postion of the server resource</param>
+        [ObserversRpc]
+        public virtual void SetNewPostion(Vector3 serverPosition) => transform.position = serverPosition;
```

Comment:
```suggestion
        public virtual void SetNewPostion(Vector3 newPosition) => transform.position = serverPosition;
```
a bit more clear than `serverPosition` because when I read `serverPosition` I am expecting it to come from the server and not to be send to other clients